### PR TITLE
test: adds integration test for x-goog-user-project header

### PR DESCRIPTION
**Pull request by bcoe**
Adds integration test for the population of x-goog-user-project, demonstrating that our patching of google-auth-library-nodejs is working for gRPC libraries.

We've opted to add the test here rather than in `nodejs-logging`, because this header only has an effect on non-resourced APIs.

### CODE REVIEW

File: system-test/translate.ts [hunk 1]
```diff
@@ -135,5 +136,48 @@ describe('translate', () => {
...
+      assert.deepStrictEqual(englishResult, {
+        languageCode: 'en',
+        displayName: '',
+        supportSource: true,
+        supportTarget: true,
+      });
+
+      // Ensure we actually populated the header:
+      assert.strictEqual(
+        'long-door-651',
```
Comment (broady) [inline system-test/translate.ts]: this will fail locally, i think? and only pass on kokoro.

File: system-test/translate.ts [hunk 2]
```diff
@@ -135,5 +136,48 @@ describe('translate', () => {
...
+        (l: {[key: string]: string}) => {
+          return l.languageCode === 'en';
+        }
+      )[0];
+      assert.deepStrictEqual(englishResult, {
+        languageCode: 'en',
+        displayName: '',
+        supportSource: true,
+        supportTarget: true,
+      });
```
Comment (jadekler) [inline system-test/translate.ts]: nit: unnecessary for this test?

File: system-test/translate.ts (hunk 1 above)
Comment (bcoe) [reply system-test/translate.ts]: shoot, this should be an environment variable good catch.

File: system-test/translate.ts (hunk 2 above)
Comment (bcoe) [reply system-test/translate.ts]: seems reasonable to remove, I think it's enough for us to not throw an exception.

**Comment by bcoe** [comment]
@broady this does hit a slightly different code path than relying on Account Default Credentials would, but I think it's a worthwhile tradeoff for the tests being significantly easier to setup. I'm fairly confident in the unit tests we have in the upstream `google-auth-library-nodejs` at this point, and I mostly wanted to prove with this API that `@grpc/grpc-js` is actually populating headers as expected.

**Pull request merged**